### PR TITLE
Move monorepo tests into sub-project

### PR DESCRIPTION
**Pull request by evantahler**
This PR moves our internal monorepo-level tests to a subproject.  This will reduce the dependencies required at the root-level of the project, and help make staging deploys that /have/ to run `npm install` work again maybe

### Development Notes

I'm doing local testing against sentry.  

1. `roo demo -cl` - make some data with multiple sources and a destination
2. enable sentry for EVERY task by setting `SENTRY_DSN=-xxx` and `SENTRY_TRACE_SAMPLE_RATE=1.0`
3. `roo run`

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

**Comment by evantahler** [comment]
Here's the new test running OK - https://app.circleci.com/pipelines/github/grouparoo/grouparoo/15314/workflows/3e4cb225-a789-46c9-b3c9-2c9f5ecd3d9e/jobs/430622

File: monorepo/__tests__/packages.ts [hunk 1]
```diff
@@ -110,8 +116,13 @@ describe("package.json validation", () => {
     childProcess.exec(
       `./node_modules/.bin/syncpack list-mismatches`,
       { cwd: path.resolve(__dirname + "/../") },
-      (err, stdout) => {
-        done(stdout);
+      (err, stdout, stderr) => {
+        console.log("[syncpack]: " + stdout);
+        if (err || stderr) {
+          console.error(err || stderr);
+          throw new Error("syncpack error");
```
Comment (teallarson) [inline monorepo/__tests__/packages.ts:123]: Note that the useful information (which packages are mismatched) will be in `stdout` not `err` or `stderr`.

File: package.json [hunk 2]
```diff
@@ -13,27 +13,15 @@
...
-    "license-checker": "github:grouparoo/license-checker",
-    "npm-check-updates": "12.2.1",
-    "prettier": "2.5.1",
-    "syncpack": "6.2.0",
-    "ts-jest": "27.1.3"
+    "npm-check-updates": "12.2.1"
   },
   "scripts": {
     "preinstall": "./bin/preinstall",
-    "version-fixer": "./tools/version-checker/fix",
```
Comment (teallarson) [inline package.json:31]: This means we don't currently have anywhere to call this from if the version-checker test is failing.  Do we want this somewhere as a script?

**Review by teallarson — APPROVED** [review]
Let's try it!  A few comments but none blocking.

File: package.json (hunk 2 above)
Comment (evantahler) [reply package.json:31]: You can still run `./tools/version-checker/fix` yourself, just not from pnpm.  It seemed a little out of place to me 🤷

File: monorepo/__tests__/packages.ts (hunk 1 above)
Comment (evantahler) [reply monorepo/__tests__/packages.ts:123]: ah-ha! I'll add that back.

Comment (evantahler) [reply monorepo/__tests__/packages.ts:123]: Actually, it seems to be OK as-is.  I changed one of the versions of jest and the test failed:


```
@evantahler ➜ /workspaces/grouparoo/monorepo (main ✗) $ npm test
> @grouparoo/monorepo@0.0.0 test
> jest

  console.log
    [syncpack]: - jest 27.5.1
      27.5.0 in devDependencies of @grouparoo/monorepo
      27.5.1 in devDependencies of @grouparoo/core
      27.5.1 in devDependencies of @grouparoo/app-staging-community
      27.5.1 in devDependencies of @grouparoo/app-staging-config
      27.5.1 in devDependencies of @grouparoo/app-staging-enterprise
      27.5.1 in devDependencies of @grouparoo/airtable
      27.5.1 in devDependencies of @grouparoo/app-templates
      27.5.1 in devDependencies of @grouparoo/bigquery
      27.5.1 in devDependencies of @grouparoo/braze
      27.5.1 in devDependencies of @grouparoo/calculated-property
      27.5.1 in devDependencies of @grouparoo/clickhouse
      27.5.1 in devDependencies of @grouparoo/closeio
      27.5.1 in devDependencies of @grouparoo/cloudwatch
      27.5.1 in devDependencies of @grouparoo/csv
      27.5.1 in devDependencies of @grouparoo/customerio
      27.5.1 in devDependencies of @grouparoo/dbt
      27.5.1 in devDependencies of @grouparoo/demo
      27.5.1 in devDependencies of @grouparoo/eloqua
      27.5.1 in devDependencies of @grouparoo/facebook
      27.5.1 in devDependencies of @grouparoo/google-sheets
      27.5.1 in devDependencies of @grouparoo/hubspot
      27.5.1 in devDependencies of @grouparoo/intercom
      27.5.1 in devDependencies of @grouparoo/iterable
      27.5.1 in devDependencies of @grouparoo/klaviyo
      27.5.1 in devDependencies of @grouparoo/logger
      27.5.1 in devDependencies of @grouparoo/mailchimp
      27.5.1 in devDependencies of @grouparoo/mailjet
      27.5.1 in devDependencies of @grouparoo/marketo
      27.5.1 in devDependencies of @grouparoo/mixpanel
      27.5.1 in devDependencies of @grouparoo/mongo
      27.5.1 in devDependencies of @grouparoo/mysql
      27.5.1 in devDependencies of @grouparoo/newrelic
      27.5.1 in devDependencies of @grouparoo/onesignal
      27.5.1 in devDependencies of @grouparoo/pardot
      27.5.1 in devDependencies of @grouparoo/pipedrive
      27.5.1 in devDependencies of @grouparoo/postgres
      27.5.1 in devDependencies of @grouparoo/prometheus
      27.5.1 in devDependencies of @grouparoo/redshift
      27.5.1 in devDependencies of @grouparoo/rollbar
      27.5.1 in devDependencies of @grouparoo/sailthru
      27.5.1 in devDependencies of @grouparoo/salesforce
      27.5.1 in devDependencies of @grouparoo/sendgrid
      27.5.1 in devDependencies of @grouparoo/sentry
      27.5.1 in devDependencies of @grouparoo/snowflake
      27.5.1 in devDependencies of @grouparoo/spec-helper
      27.5.1 in devDependencies of @grouparoo/sqlite
      27.5.1 in devDependencies of @grouparoo/vero
      27.5.1 in devDependencies of @grouparoo/zendesk
      27.5.1 in devDependencies of @grouparoo/ui-community
      27.5.1 in devDependencies of @grouparoo/ui-components
      27.5.1 in devDependencies of @grouparoo/ui-config
      27.5.1 in devDependencies of @grouparoo/ui-enterprise

      at __tests__/packages.ts:120:17

  console.error
    Error: Command failed: ./node_modules/.bin/syncpack list-mismatches
    
        at ChildProcess.exithandler (node:child_process:397:12)
        at ChildProcess.emit (node:events:390:28)
        at maybeClose (node:internal/child_process:1064:16)
        at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {
      killed: false,
      code: 1,
      signal: null,
      cmd: './node_modules/.bin/syncpack list-mismatches'
    }

      120 |         console.log("[syncpack]: " + stdout);
      121 |         if (err || stderr) {
    > 122 |           console.error(err || stderr);
          |                   ^
      123 |           throw new Error("syncpack error");
      124 |         }
      125 |         done();

      at __tests__/packages.ts:122:19
```

Comment (teallarson) [reply monorepo/__tests__/packages.ts:123]: > Actually, it seems to be OK as-is. I changed one of the versions of jest and the test failed:
> 
> ```
> @evantahler ➜ /workspaces/grouparoo/monorepo (main ✗) $ npm test
> > @grouparoo/monorepo@0.0.0 test
> > jest
> 
>   console.log
>     [syncpack]: - jest 27.5.1
>       27.5.0 in devDependencies of @grouparoo/monorepo
>       27.5.1 in devDependencies of @grouparoo/core
>       27.5.1 in devDependencies of @grouparoo/app-staging-community
>       27.5.1 in devDependencies of @grouparoo/app-staging-config
>       27.5.1 in devDependencies of @grouparoo/app-staging-enterprise
>       27.5.1 in devDependencies of @grouparoo/airtable
>       27.5.1 in devDependencies of @grouparoo/app-templates
>       27.5.1 in devDependencies of @grouparoo/bigquery
>       27.5.1 in devDependencies of @grouparoo/braze
>       27.5.1 in devDependencies of @grouparoo/calculated-property
>       27.5.1 in devDependencies of @grouparoo/clickhouse
>       27.5.1 in devDependencies of @grouparoo/closeio
>       27.5.1 in devDependencies of @grouparoo/cloudwatch
>       27.5.1 in devDependencies of @grouparoo/csv
>       27.5.1 in devDependencies of @grouparoo/customerio
>       27.5.1 in devDependencies of @grouparoo/dbt
>       27.5.1 in devDependencies of @grouparoo/demo
>       27.5.1 in devDependencies of @grouparoo/eloqua
>       27.5.1 in devDependencies of @grouparoo/facebook
>       27.5.1 in devDependencies of @grouparoo/google-sheets
>       27.5.1 in devDependencies of @grouparoo/hubspot
>       27.5.1 in devDependencies of @grouparoo/intercom
>       27.5.1 in devDependencies of @grouparoo/iterable
>       27.5.1 in devDependencies of @grouparoo/klaviyo
>       27.5.1 in devDependencies of @grouparoo/logger
>       27.5.1 in devDependencies of @grouparoo/mailchimp
>       27.5.1 in devDependencies of @grouparoo/mailjet
>       27.5.1 in devDependencies of @grouparoo/marketo
>       27.5.1 in devDependencies of @grouparoo/mixpanel
>       27.5.1 in devDependencies of @grouparoo/mongo
>       27.5.1 in devDependencies of @grouparoo/mysql
>       27.5.1 in devDependencies of @grouparoo/newrelic
>       27.5.1 in devDependencies of @grouparoo/onesignal
>       27.5.1 in devDependencies of @grouparoo/pardot
>       27.5.1 in devDependencies of @grouparoo/pipedrive
>       27.5.1 in devDependencies of @grouparoo/postgres
>       27.5.1 in devDependencies of @grouparoo/prometheus
>       27.5.1 in devDependencies of @grouparoo/redshift
>       27.5.1 in devDependencies of @grouparoo/rollbar
>       27.5.1 in devDependencies of @grouparoo/sailthru
>       27.5.1 in devDependencies of @grouparoo/salesforce
>       27.5.1 in devDependencies of @grouparoo/sendgrid
>       27.5.1 in devDependencies of @grouparoo/sentry
>       27.5.1 in devDependencies of @grouparoo/snowflake
>       27.5.1 in devDependencies of @grouparoo/spec-helper
>       27.5.1 in devDependencies of @grouparoo/sqlite
>       27.5.1 in devDependencies of @grouparoo/vero
>       27.5.1 in devDependencies of @grouparoo/zendesk
>       27.5.1 in devDependencies of @grouparoo/ui-community
>       27.5.1 in devDependencies of @grouparoo/ui-components
>       27.5.1 in devDependencies of @grouparoo/ui-config
>       27.5.1 in devDependencies of @grouparoo/ui-enterprise
> 
>       at __tests__/packages.ts:120:17
> 
>   console.error
>     Error: Command failed: ./node_modules/.bin/syncpack list-mismatches
>     
>         at ChildProcess.exithandler (node:child_process:397:12)
>         at ChildProcess.emit (node:events:390:28)
>         at maybeClose (node:internal/child_process:1064:16)
>         at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5) {
>       killed: false,
>       code: 1,
>       signal: null,
>       cmd: './node_modules/.bin/syncpack list-mismatches'
>     }
> 
>       120 |         console.log("[syncpack]: " + stdout);
>       121 |         if (err || stderr) {
>     > 122 |           console.error(err || stderr);
>           |                   ^
>       123 |           throw new Error("syncpack error");
>       124 |         }
>       125 |         done();
> 
>       at __tests__/packages.ts:122:19
> ```

Oh awesome!

**Pull request merged**